src/views/ProductView.vue

```diff
@@ -744,173 +750,271 @@ function goToWishlists() {
-            (productItem) =>
-                productItem.inventoryStatus === ProductItemStatus.Shipped
-        ).length
-    }
+const productItemsCountOfReturnedItems = asyncComputed(
+    async () => {
+        trigger.value
 
-    return undefined
-})
+        const queryResponse = await getProductItemsCountOfInventoryStatus(
+            ProductItemStatus.Returned
+        )
+
+        if (queryResponse != undefined) {
+            return queryResponse.productItems.totalCount
+        } else {
+            return undefined
+        }
+    },
+    undefined,
+    {
+        onError: (e) =>
+            pushErrorNotification(
+                errorMessages.getProductItemsCountOfInventoryStatus,
+                e
+            ),
+        shallow: false,
+    }
+)
 
 /**
- * Computed property for the number of delivered product items.
- * @returns The number of delivered product items or undefined if no data is available.
+ * Asynchronously computed property for the number of lost product items.
+ * @returns The number of lost product items or undefined if no data is available.
  */
-const numberOfDeliveredProductItems = computed(() => {
-    if (inventoryStatusOfProductItems.value !== null) {
-        return inventoryStatusOfProductItems.value.filter(
-            (productItem) =>
-                productItem.inventoryStatus === ProductItemStatus.Delivered
-        ).length
-    }
+const productItemsCountOfLostItems = asyncComputed(
+    async () => {
+        trigger.value
 
-    return undefined
-})
+        const queryResponse = await getProductItemsCountOfInventoryStatus(
+            ProductItemStatus.Lost
+        )
+
+        if (queryResponse != undefined) {
+            return queryResponse.productItems.totalCount
+        } else {
+            return undefined
+        }
+    },
+    undefined,
+    {
+        onError: (e) =>
+            pushErrorNotification(
+                errorMessages.getProductItemsCountOfInventoryStatus,
+                e
+            ),
+        shallow: false,
+    }
+)
 
 /**
- * Computed property for the number of returned product items.
- * @returns The number of returned product items or undefined if no data is available.
+ * Asynchronously gets the count of product items with a specified inventory status.
+ * @param inventoryStatus - The inventory status of the product items.
+ * @returns A promise that resolves to the query result containing the count of product items, or null if retrieval is not possible.
  */
-const numberOfReturnedProductItems = computed(() => {
-    if (inventoryStatusOfProductItems.value !== null) {
-        return inventoryStatusOfProductItems.value.filter(
-            (productItem) =>
-                productItem.inventoryStatus === ProductItemStatus.Returned
-        ).length
+async function getProductItemsCountOfInventoryStatus(
+    inventoryStatus: ProductItemStatus
+): Promise<GetProductItemsCountOfInventoryStatusQuery | null> {
+    if (inventoryStatus == undefined) {
```

Comment:
Not sure about Typescript, but it this the best way to solve this? How about an Optional?

Reply:
This is the way you define optionals as return types in TS. You can use the '?' syntax for defining members of interfaces or classes but not for the return types.